user/serializers.py

```diff
@@ -1,18 +1,235 @@
+
+    def validate(self, validated_data):
+        emails = [email.strip().lower() for email in validated_data["emails"].split(",")]
+        registered_emails = User.objects.filter(email__in=emails).values_list("email", flat=True)
+        new_emails = list(set(emails) - set(registered_emails))
+        if not new_emails:
+            raise serializers.ValidationError("All emails are already registered. provide new emails")
+        validated_data["new_emails"] = new_emails
+        return validated_data
+
+    def create(self, validated_data):
+        new_emails = validated_data.get("new_emails")
+        new_users = []
+        with transaction.atomic():
+            for email in new_emails:
+                user = User.objects.create_user(
+                    email=email,
+                    password=None,
+                    is_active=False,
+                )
+                transaction.on_commit(lambda user_id=user.id: send_account_creation_email_task.delay(user_id))
+                new_users.append(user)
+        return new_users
+
+
+class UserRegisterSerializer(serializers.ModelSerializer):
+    uuid = serializers.CharField(required=True)
+    token = serializers.CharField(required=True)
+    confirm_password = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = User
+        fields = ("uuid", "token", "password", "confirm_password", "first_name", "last_name", "department")
+        extra_kwargs = {"password": {"write_only": True}}
+
+    def validate_password(self, password):
+        validate_password(password)
+        return password
+
+    def validate(self, attrs):
+        if attrs["password"] != attrs["confirm_password"]:
+            raise serializers.ValidationError(gettext("Passwords do not match."))
+        return {**attrs, "user": validate_token(attrs, TokenManager.account_registration_token_generator)}
+
+    def save(self, **_):
+        assert isinstance(self.validated_data, dict)
+        user = self.validated_data["user"]
+        user.first_name = self.validated_data["first_name"]
+        user.last_name = self.validated_data["last_name"]
+        user.department = self.validated_data["department"]
+        user.set_password(self.validated_data["password"])
+        user.is_active = True
+        user.save()
+        return user
+
+
+class UserResendInviteSerializer(serializers.Serializer):
+    user_id = serializers.CharField(required=True)
+
+    def validate(self, attrs):
+        user_id = attrs["user_id"]
+        user = get_object_or_404(User, id=user_id)
+        if user.is_active:
+            raise serializers.ValidationError(gettext("User is already active."))
+        return {
+            **attrs,
+            "user": user,
+        }
+
+    def save(self, **_):
+        assert isinstance(self.validated_data, dict)
+        user_id = self.validated_data["user_id"]
+        resend_account_activation_task.delay(user_id)
```

Comment:
Use transaction here

```suggestion
        transaction.on_commit(lambda user_id=user.id: resend_account_activation_task.delay(user_id))
```